aff4/volume_group.h

```diff
@@ -2,17 +2,22 @@
 #define     AFF4_VOLUME_GROUP_H_
 
 #include <unordered_map>
+#include <set>
+
 #include "aff4/aff4_io.h"
 #include "aff4/data_store.h"
 
-
-
 namespace aff4 {
 
  class VolumeGroup {
  protected:
-     std::unordered_map<URN, AFF4Flusher<AFF4Volume>> volume_objs;
-     DataStore *resolver;
+    std::unordered_map<URN, AFF4Flusher<AFF4Volume>> volume_objs;
+    DataStore *resolver;
+
+    std::set<std::string> searchPaths;
+    std::unordered_map<URN, std::string> foundVolumes;
```

Comment:
Why keep a special map instead of the resolver? You can do the same thing within the existing resolver and then anything else can know about the same volumes.